packages/components/src/components/hds/icon/index.ts

```diff
@@ -112,4 +150,38 @@ export default class HdsIcon extends Component<HdsIconSignature> {
 
     return classes.join(' ');
   }
+
+  get viewBox(): string {
+    const { name } = this.args;
+    const registryEntry = IconRegistry[name as HdsIconName];
+
+    if (this.hdsCarbon.carbonModeEnabled && registryEntry?.carbon) {
+      return '0 0 32 32';
+    }
+
+    return `0 0 ${this.size} ${this.size}`;
+  }
+
+  loadIconModuleTask = task(
+    async (name: IconName, size: HdsIconSizes, carbonModeEnabled: boolean): Promise<void> => {
+      const registryEntry = IconRegistry[name as HdsIconName];
+
+      assert(
+        `The icon @name "${name}" or @size "${size}" provided to <Hds::Icon> is not correct.`,
+        registryEntry !== undefined
+      );
+
+      const shouldUseCarbon = carbonModeEnabled && registryEntry.carbon;
+      
+      const loader = shouldUseCarbon 
```

Comment:
no need to define a temporary variable, we can use directly the evaluation in the ternary
```suggestion
      const loader = carbonModeEnabled && registryEntry.carbon
```